include/grpc++/impl/codegen/thrift_serializer.h

```diff
@@ -131,12 +131,12 @@ class ThriftSerializer {
     last_deserialized_ = true;
 
     // reset buffer transport
-    buffer_->resetBuffer((uint8_t*)serialized_buffer, length);
+    buffer_->resetBuffer(const_cast<uint8_t*>(serialized_buffer), length);
 
     // read the protocol version if necessary
     if (serialize_version_) {
       std::string name = "";
-      TMessageType mt = (TMessageType)0;
+      TMessageType mt = static_cast<TMessageType>(0);
```

Comment:
serialization version doesn't have any Message Type that is why it is set
to  0.

On Thu, Aug 4, 2016 at 5:03 PM, Vijay Pai <notifications@github.com> wrote:

> In include/grpc++/impl/codegen/thrift_serializer.h
> <https://github.com/grpc/grpc/pull/7594#discussion_r73623829>:
>
> >
> >      // read the protocol version if necessary
> >      if (serialize_version_) {
> >        std::string name = "";
> > -      TMessageType mt = (TMessageType)0;
> > +      TMessageType mt = static_cast<TMessageType>(0);
>
> But then why aren't you using the actual enum typename rather than a
> static cast from an int?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/grpc/grpc/pull/7594/files/2e698457f370091111d0fe620bad306c743a3548..76db7f9a7cdf6b5fd6706037c72de1dd131b181e#r73623829>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ASfuIpZ8JE-GXPQCyLb81_LebwjNg9I3ks5qcn3cgaJpZM4JZPv_>
> .
>